### PR TITLE
ctld: plug memory leaks

### PR DESCRIPTION
**Pull request by asomers**
MFC after:	2 weeks
Sponsored by:	Axcient
Reported by:	valgrind

### CODE REVIEW

File: usr.sbin/ctld/kernel.c [hunk 1]
```diff
@@ -614,6 +614,23 @@ conf_new_from_kernel(void)
 		}
 		cp->p_ctl_port = port->port_id;
 	}
+	while ((port = STAILQ_FIRST(&devlist.port_list))) {
+		struct cctl_lun_nv *nv;
+
+		port = STAILQ_FIRST(&devlist.port_list);
+		STAILQ_REMOVE_HEAD(&devlist.port_list, links);
```
Comment (amotin) [inline usr.sbin/ctld/kernel.c:621]: Why duplicate STAILQ_FIRST()?

File: usr.sbin/ctld/kernel.c [hunk 2]
```diff
@@ -664,6 +681,19 @@ conf_new_from_kernel(void)
 				    cl->l_name);
 		}
 	}
+	while ((lun = STAILQ_FIRST(&devlist.lun_list))) {
+		struct cctl_lun_nv *nv;
+
+		lun = STAILQ_FIRST(&devlist.lun_list);
+		STAILQ_REMOVE_HEAD(&devlist.lun_list, links);
```
Comment (amotin) [inline usr.sbin/ctld/kernel.c:688]: Again duplicate STAILQ_FIRST().

Comment (asomers) [reply usr.sbin/ctld/kernel.c]: fixed

**Review by amotin — APPROVED** [review]
Didn't look very deep, but seems fine.

**Comment by asomers** [comment]
Commited as  2909ddd17cb4d750852dc04128e584f93f8c5058 .

**Pull request closed without merging**